import.go

```diff
@@ -109,54 +101,131 @@ func importYamlWithTemplateFunc(r io.Reader, tmplFunc TemplateFunc) ([]*YamlSect
 		}}
 	}
 	for i := range chunks {
-		vp := viper.New()
-		vp.SetConfigType("yaml")
+		//Each chunk is a document, needs to be filtered so the consumer gets the sections they want
+		// then run the supplied template, then Vipered
+		// this avoids running templates on unrelated sections that we may not have the data for
+
+		//extract section
 		var b []byte
 		if i < len(chunks)-1 {
 			b = data[chunks[i][1]:chunks[i+1][0]]
 		} else {
 			b = data[chunks[i][1]:]
 		}
-		if err := vp.ReadConfig(bytes.NewBuffer(b)); err != nil {
-			fmt.Printf("---\n%v\n", string(b))
-			return nil, errors.Wrap(err, "failed to import yaml section")
-		}
-		sections = append(sections, &YamlSection{Bytes: b, Viper: vp})
+
+		//save completed section
+		sections = append(sections, &YamlSection{
+			Bytes:         b,
+			OriginalBytes: b,
+		})
 	}
 	return sections, nil
 }
 
-func importYaml(r io.Reader) ([]*YamlSection, error) {
-	sections := []*YamlSection{}
-	buf := new(bytes.Buffer)
-	_, err := buf.ReadFrom(r)
+func Filter(in []*YamlSection, filters []string) ([]*YamlSection, error) {
+	ret := []*YamlSection{}
+	for _, section := range in {
+		//Each section is a document, needs to be filtered so the consumer gets the sections they want
+		// then run the supplied template, then Vipered
+		// this avoids running templates on unrelated sections that we may not have the data for
+
+		//run filters
+		if !func() bool {
```

Comment:
Is there a reason we prefer this to be an anonymous function?

Reply:
It's to convey the 'continue' to the outer loop. The other way would be to use a goto label, but this seemed nicer at the time. 
Or perhaps I am missing a different structure to use?

Reply:
breaking out to functions.